examples/rate_providers/rate_providers.py

```diff
@@ -0,0 +1,128 @@
+import asyncio
+import json
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Callable, Dict, Tuple
+
+from golem_core import GolemNode, Payload, commands
+from golem_core.default_logger import DefaultLogger
+from golem_core.default_payment_manager import DefaultPaymentManager
+from golem_core.low.activity import Activity
+from golem_core.mid import (
+    Buffer,
+    Chain,
+    Limit,
+    Map,
+    default_create_activity,
+    default_create_agreement,
+    default_negotiate,
+)
+
+FRAME_CONFIG_TEMPLATE = json.loads(Path("frame_params.json").read_text())
+PAYLOAD = Payload.from_image_hash(
+    "9a3b5d67b0b27746283cb5f287c13eab1beaa12d92a9f536b747c7ae"
+)
+SUBNET = "public"
+
+# scores will contain time of profiling execution
+scores: Dict[str, timedelta] = {}
+
+
+async def delay(delay: timedelta, func: Callable):
+    await asyncio.sleep(delay.total_seconds())
+    return func()
+
+
+async def filter_rated_providers(offer_stream):
+    async for offer in offer_stream:
+        provider_id = offer.data.issuer_id
+        if provider_id not in scores:
+            scores[provider_id] = None
+            yield offer
+
+
+async def prepare_activity(activity: Activity) -> Activity:
+    print(f"Prepare {activity}")
+    batch = await activity.execute_commands(
+        commands.Deploy(),
+        commands.Start(),
+        commands.SendFile("cubes.blend", "/golem/resource/scene.blend"),
+    )
+    await batch.wait(timeout=60)
+    return activity
+
+
+async def rate_provider(activity: Activity) -> str:
+    provider_id = activity.parent.parent.data.issuer_id
+    print(f"Rating {provider_id}")
+    frame_config = FRAME_CONFIG_TEMPLATE.copy()
+    fname = f"out{0:04d}.png"  # format used by `run-blender.sh` included in `PAYLOAD`
```

Comment:
```suggestion
    fname = "out{0:04d}.png"  # format used by `run-blender.sh` included in `PAYLOAD`
```

As we're not want to call fstring right now, but later. I'm surprised that it's not raising exception due to lack of `0` variable in the scope.

Reply:
0 is a number, can't make a variable starting with a number :joy: